### PR TITLE
7 ticket

### PR DESCRIPTION
**Pull request by Hassen-Ahmed**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -155,4 +155,106 @@ describe("All requests", () => {
...
+                .expect(201)
+                .send({
+                    user_name: "butter_bridge",
+                    body: {
+                        article_id: 9,
+                        author: "butter_bridge",
+                        votes: 20,
+                        created_at: new Date(),
+                        body: "post request will going to be ok",
+                    },
```
Comment (MKD01) [inline __tests__/app.test.js:171]: For this ticket body should not be an object here, what we should expect is body to be a string containing only the comment you would like to post.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -155,4 +155,106 @@ describe("All requests", () => {
...
+                })
+                .then(({ body }) => {
+                    const { comments } = body;
+                    expect(comments).toHaveProperty("comment_id", expect.any(Number));
+                    expect(comments).toHaveProperty("body", expect.any(String));
+                    expect(comments).toHaveProperty("article_id", expect.any(Number));
```
Comment (MKD01) [inline __tests__/app.test.js:177]: In this case we actually do know what the article id should be as we are posting a comment to a specific article, so this assertion can be made more precise.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -155,4 +155,106 @@ describe("All requests", () => {
...
+                        votes: 20,
+                        created_at: new Date(),
+                        body: "post request will going to be ok",
+                    },
+                })
+                .then(({ body }) => {
+                    const { msg } = body;
+                    expect(msg).toBe("Invalid request!");
+                });
+        });
```
Comment (MKD01) [inline __tests__/app.test.js:239]: This is not a valid test as we should not be expecting article id within body (as mention in the above comment)

File: __tests__/app.test.js [hunk 4]
```diff
@@ -155,4 +155,106 @@ describe("All requests", () => {
...
+    });
```
Comment (MKD01) [inline __tests__/app.test.js:259]: This is also not a valid test.

File: __tests__/app.test.js [hunk 5]
```diff
@@ -155,4 +155,106 @@ describe("All requests", () => {
...
 });
```
Comment (MKD01) [inline __tests__/app.test.js:260]: There are still some tests that should be added here.

What should we expect if the user made a request to an invalid article id?  e.g. /api/articles/banana/comments

What should we expect if the user does not give us the required fields? e.g. misisng username or body

**Review by MKD01 — CHANGES_REQUESTED** [review]
I have added some comments for you to read through, please re-submit this pr once you have made all the requested changes.

**Comment by Hassen-Ahmed** [comment]
The instructions need some extra explanation to make right testing and now
you briefly explain to me.
Thanks a lot😀!

**Pull request closed without merging**